src/lib/math/bigint/big_ops2.cpp

```diff
@@ -11,6 +11,20 @@
 #include <botan/internal/mp_core.h>
 #include <algorithm>
 
+/* This hack works around an undefined reference to __udivti3() when compiling for 64-bit Windows
+ * using clang-cl, see:
+ *
+ * https://github.com/llvm/llvm-project/issues/25679
+ * https://stackoverflow.com/questions/68676184/clang-cl-error-lld-link-error-undefined-symbol-divti3
+ *
+ * I couldn't come up with a way to embed this into the build info files without extending
+ * configure.py to allow cpu-specific libs in the compiler info or os+cc+arch-specific libs in the
+ * module info. Hopefully this can go away when the above Clang issue is fixed anyway.
+*/
+#if defined(_MSC_VER) && defined(_WIN64) && defined(__clang__)
+   #pragma comment(lib, "clang_rt.builtins-x86_64.lib")
+#endif
```

Comment:
Unfortunately, this doesn't seem to work on my windows environment:

```
lld-link: error: undefined symbol: __udivti3
>>> referenced by botan-3.lib(math_bigint_big_ops3.obj):(unsigned __int64 __cdecl Botan::operator%(class Botan::BigInt const &, unsigned __int64))
>>> referenced by botan-3.lib(math_bigint_big_ops2.obj):(public: unsigned __int64 __cdecl Botan::BigInt::operator%=(unsigned __int64))
>>> referenced by botan-3.lib(math_bigint_divide.obj):(void __cdecl Botan::vartime_divide(class Botan::BigInt const &, class Botan::BigInt const &, class Botan::BigInt &, class Botan::BigInt &))
ninja: build stopped: subcommand failed.
```

Configuring/Building like:

```
 ./configure --without-documentation --cc=clangcl --disable-shared-library --build-tool=ninja --compiler-cache=sccache --build-targets=static,tests
ninja tests
```

Reply:
@reneme

How are you running the configure script? I'm using `python configure.py ...` and then `ninja` from the "x64 Native Tools Command Prompt for VS2022", I checked out your branch and that builds fine for me too.

Here's my compiler version, for comparison:

![image](https://github.com/user-attachments/assets/4e32187d-4c3a-41b9-a60b-8176e76011f5)

Reply:
It builds in CI now, although its failing on something to do with installing afterwards.

Reply:
The patch below should fix the installation:

```patch
diff --git a/src/scripts/ci_check_install.py b/src/scripts/ci_check_install.py
index 5b4ec18f1..f6a695115 100755
--- a/src/scripts/ci_check_install.py
+++ b/src/scripts/ci_check_install.py
@@ -27,7 +27,7 @@ def verify_library(build_config):

     major_version = int(build_config["version_major"])

-    if build_config['compiler'] == 'msvc':
+    if build_config['compiler'] in ['msvc', 'clangcl']:
         expected_lib_format = r'^botan-%d\.(dll|lib)$' % (major_version)
     elif build_config['os'] == 'macos':
         expected_lib_format = r'^libbotan-%d\.(a|dylib)$' % (major_version)
```

Reply:
For reasons unknown my clang_rt doesn't seem to provide this builtin.

```
MEUSEL@BS001125 C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.38.33130\lib\x64
(0) > dumpbin /symbols .\clang_rt.builtins-x86_64.lib | Select-String "udiv"

00D 00000000 SECT4  notype ()    External     | __udivsi3
00D 00000000 UNDEF  notype ()    External     | __udivsi3
00E 00000000 SECT4  notype ()    External     | __udivmodsi4
012 00000000 SECT6  notype       Static       | $unwind$__udivmodsi4
015 00000000 SECT7  notype       Static       | $pdata$__udivmodsi4
00D 00000000 SECT4  notype ()    External     | __udivmoddi4
011 00000000 SECT6  notype       Static       | $unwind$__udivmoddi4
014 00000000 SECT7  notype       Static       | $pdata$__udivmoddi4
00D 00000000 SECT4  notype ()    External     | __udivdi3
00D 00000000 UNDEF  notype ()    External     | __udivmoddi4
00D 00000000 UNDEF  notype ()    External     | __udivmodsi4
00D 00000000 UNDEF  notype ()    External     | __udivmoddi4
00E 00000000 UNDEF  notype ()    External     | __udivmoddi4
```

Reply:
Never mind. Updating Visual Studio helped.